### DISCUSSION TITLE
Query result to Python flavor

### DISCUSSION THREAD

**Discussion by benshaji-sequoiaat** [react-querybuilder/react-querybuilder#578] [General]
I was searching for something which can take the query generated from our query-builder and convert them to Python flavor. For example, consider below Json query:
```
{
  "id": "32f8293f-a9cc-43ea-9e30-54106ea538a2",
  "combinator": "and",
  "not": false,
  "rules": [
    {
      "id": "33137f3b-8a5b-4c78-94dd-cb273ce8c1fe",
      "field": "firstName",
      "value": "Stev",
      "operator": "="
    },
    {
      "id": "269cdc50-5349-4f2e-9167-b209e3b4ec55",
      "field": "lastName",
      "operator": "=",
      "valueSource": "value",
      "value": "KM"
    },
    {
      "id": "13c39b66-3d61-4122-8873-b8825c237188",
      "rules": [
        {
          "id": "1ffcc23b-e2ca-4e8a-9fe8-8a71705f219f",
          "field": "firstName",
          "operator": "=",
          "valueSource": "value",
          "value": "Ben"
        },
        {
          "id": "80764836-5375-4f94-a3fb-d2865e80377e",
          "field": "height",
          "operator": "=",
          "valueSource": "value",
          "value": "180"
        }
      ],
      "combinator": "or",
      "not": false
    }
  ]
}
```  

To Python as:

```
(first_name == "Stev" and last_name == "KM") or (first_name == "Ben" and height == "180")
```

I know we have nothing to do with Python or a parser like i'm asking here. Still, If someone knows something can do this parsing, i hope you can reply to this post. Other ideas/ suggestions are also welcome :)

**Comment by jakeboone02** [comment]
`formatQuery` is the official "export" method ([docs here](https://react-querybuilder.js.org/docs/utils/export)). The available formats are hard-coded, but we can certainly look at adding new ones.

Another option would be to write your own method based on one of the existing formats. In your case I would recommend looking at the code for the CEL or SpEL formats ([here](https://github.com/react-querybuilder/react-querybuilder/blob/b9bf29832c942bbfdc87e5b7df95fcfcc6772fbe/packages/react-querybuilder/src/utils/formatQuery/formatQuery.ts#L475) and [here](https://github.com/react-querybuilder/react-querybuilder/blob/b9bf29832c942bbfdc87e5b7df95fcfcc6772fbe/packages/react-querybuilder/src/utils/formatQuery/formatQuery.ts#L520), respectively).

**Reply by benshaji-sequoiaat** [reply]
Hi, thanks for the reply.

It seems the SQL output seems very close to Python syntax. For example;  `"Fuel Level" = '50' AND "Start" = '888' AND ("Coolant Temp" > '66' OR "Amps A" <= '66')` can be converted to Python as `"Fuel Level" == '50' and "Start" == '888' and ("Coolant Temp" > '66' or "Amps A" <= '66')`. 

I guess, converting from Json (from query builder) to Python seems bit complex than from SQL as above example says. 

Thanks for your suggestion.
